### PR TITLE
Update DG diagrams

### LINKED ISSUE #251: Inaccurate diagrams
The diagrams used for DG documentation don't reflect the current implementation

### PR DESCRIPTION
**Pull request by Vishwanth2210**
DG Diagrams: Not updated to reflect the current implementation

Let's,
* Update the diagrams in `.puml` files, to reflect the correct implementation
* Save the updated diagrams as images for use in DG
* Add user stories for `undo`, `redo` and `salary`

Fixes #251

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -185,7 +182,7 @@ The `Model` component,
 
 <div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
 
-❗️ IMAGE NEEDS TO BE UPDATED
+❗️ IMAGE NEEDS TO BE UPDATED (**Remove caution if image is sufficient**)
```
Comment (AD-NAP) [inline docs/DeveloperGuide.md:185]: Do you want to add that ScheduledMeeting is made out of MeetingDate and MeetingTime?

**Review by AD-NAP — CHANGES_REQUESTED** [review]
Everything looks good. Just one minor issue - would like your opinion if we should show the OOP if ScheduledMeeting object

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Vishwanth2210) [reply docs/DeveloperGuide.md:185]: Sure, that sounds good. But I am not sure if that is giving too much details as, if I am not wrong, this diagram is catered to show the ` UniqueTaglist`. But we can include it if you want. 
If you want, I can include those in the `ModelClassDiagram` instead, as you are right, the relationship between `ScehduledMeeting`, `MeetingDate` and `MeetingTime` is not clearly shown in our diagrams.

Comment (AD-NAP) [reply docs/DeveloperGuide.md:185]: Yup, I think if it's captured in at least one, it would be good enough 👍🏽

**Review by AD-NAP — APPROVED** [review]

**Pull request merged**